src/services/firebase/firebaseClient.types.ts

```diff
@@ -8,17 +8,16 @@ export type IngredientDoc = WithKey & {
   unit: string;
 };
 
-export type PreparationMethodDoc = WithKey & {
-  description: string;
-  order: number;
-};
-
 export type RecipeDoc = WithKey & {
   description: string;
   image_url: string;
   tips: string;
   title: string;
-  yields: string;
+  yields: number;
   user_id: string;
   favorite: boolean;
+  preparation_time: number;
+  difficulty: string;
+  ingredients: IngredientDoc[];
+  preparation_method: string[];
```

Comment:
`preparation_methods` ?

I'm also wondering why are you using snake_case in firebase.